src/polkadot/polkadotEngine.ts

```diff
@@ -416,9 +428,17 @@ export class PolkadotEngine extends CurrencyEngine<PolkadotTools> {
 
     if (this.keypair == null) {
       this.keypair = new Keyring({ ss58Format: 0 })
-      this.keypair.addFromUri(
-        this.walletInfo.keys[`${this.currencyInfo.pluginId}Mnemonic`]
-      )
+      if (dotPrivateKeys.mnemonic != null) {
+        this.keypair.addFromUri(dotPrivateKeys.mnemonic)
+      } else {
+        const byteArray = []
+        for (let i = 0; i < dotPrivateKeys.privateKey.length; i += 2) {
+          const hexPair = dotPrivateKeys.privateKey.substr(i, 2)
+          byteArray.push(parseInt(hexPair, 16))
+        }
+        const uint8Array = Uint8Array.from(byteArray)
```

Comment:
Is this a hex to Unit8Array conversion? Then use `import base16 from 'rfc4648'` / `base16.parse(hex)`